aiohttp/http_writer.py

```diff
@@ -138,29 +139,36 @@ def __init__(self, stream, loop, acquire=True):
         self.output_size = 0
 
         self._eof = False
-        self._buffer = []
         self._compress = None
         self._drain_waiter = None
 
         if self._stream.available:
             self._transport = self._stream.transport
+            self._buffer = None
```

Comment:
I think it's not good idea to jungle variable type. How empty buffer `[]` is different from `None`?

Reply:
I'm using `[]` and `None` a bit differently here: once the buffer has been emptied into the new transport, it should never be used again. So I would prefer setting it to None and crash if something very wrong happens

Reply:
Hm...I see. Well, then such crash shold be controlled, imho. Otherwise it would be hard to distinguish transport re-use case from else error.

Reply:
It's purely internal to PayloadWriter, so everything is controlled where it should be.

This None means that at any given time, exactly one of `(self._transport, self._buffer)` is None. It makes it easier to check the code / prove that it's correct.